enclave/core/debugmalloc.c

```diff
@@ -399,34 +398,41 @@ void* oe_debug_realloc(void* ptr, size_t size)
 
 void* oe_debug_memalign(size_t alignment, size_t size)
 {
-    const size_t padding_size = _get_padding_size(alignment);
-    const size_t block_size = _calculate_block_size(alignment, size);
-    void* block;
-    header_t* header;
-
-    if (!(block = oe_nodebug_memalign(alignment, block_size)))
-        return NULL;
+    void* ptr = NULL;
 
-    header = (header_t*)((uint8_t*)block + padding_size);
-
-    INIT_BLOCK(header, alignment, size);
-    _check_block(header);
-    _list_insert(&_list, header);
+    // The only difference between posix_memalign and the obsolete memalign is
+    // that posix_memalign requires alignment to be a multiple of sizeof(void*).
+    // Adjust the alignment if needed.
+    if (alignment < sizeof(void*))
```

Comment:
Shouldn't this be `if (alignment % sizeof(void*) != 0) / * round up to aligned */`?

Reply:
Made it more readable using round to multiple